### ISSUE TITLE
Wayland compilation warning

### ISSUE THREAD

**Issue by Albrecht-S** [fltk/fltk#706]
Current Git, commit aebf2fec39640e2151d818e96f182af1cb3a63b3, issues a warning when compiling, using default `configure/make`. The same warning is visible in CMake builds, using `-Wall`.
- Linux Mint 20
- gcc 9.4.0

To reproduce:
```
make distclean
make [-j4]
...
drivers/Wayland/Fl_Wayland_Window_Driver.cxx: In member function ‘int Fl_Wayland_Window_Driver::set_cursor_4args(const Fl_RGB_Image*, int, int, bool)’:
drivers/Wayland/Fl_Wayland_Window_Driver.cxx:1515:47: warning: declaration ‘struct wld_window::custom_cursor’ does not declare anything
 1515 |   xid->custom_cursor = new struct wld_window::custom_cursor;
      |                                               ^~~~~~~~~~~~~
```

@ManoloFLTK It seems as if this is caused by using the same name for the struct (declaration) and the pointer variable in the original declaration of the struct. The following patch fixes the warning by renaming the struct with trailing underscore but I'm not sure if this is what you wanted.
```diff
diff --git a/src/drivers/Wayland/Fl_Wayland_Window_Driver.H b/src/drivers/Wayland/Fl_Wayland_Window_Driver.H
index 59df79f41..9ab0807aa 100644
--- a/src/drivers/Wayland/Fl_Wayland_Window_Driver.H
+++ b/src/drivers/Wayland/Fl_Wayland_Window_Driver.H
@@ -144,7 +144,7 @@ struct wld_window {
     struct xdg_popup *xdg_popup;
     struct xdg_toplevel *xdg_toplevel;
   };
-  struct custom_cursor {
+  struct custom_cursor_ {
     struct wl_cursor *wl_cursor;
     const Fl_RGB_Image *rgb;
     int hotx, hoty;
diff --git a/src/drivers/Wayland/Fl_Wayland_Window_Driver.cxx b/src/drivers/Wayland/Fl_Wayland_Window_Driver.cxx
index 969deb485..698d58088 100644
--- a/src/drivers/Wayland/Fl_Wayland_Window_Driver.cxx
+++ b/src/drivers/Wayland/Fl_Wayland_Window_Driver.cxx
@@ -79,7 +79,7 @@ Fl_Wayland_Window_Driver::Fl_Wayland_Window_Driver(Fl_Window *win) : Fl_Window_D
 }
 
 void Fl_Wayland_Window_Driver::delete_cursor_(struct wld_window *xid, bool delete_rgb) {
-  struct wld_window::custom_cursor *custom = xid->custom_cursor;
+  struct wld_window::custom_cursor_ *custom = xid->custom_cursor;
   if (custom) {
     struct wl_cursor *wl_cursor = custom->wl_cursor;
     struct cursor_image *new_image = (struct cursor_image*)wl_cursor->images[0];
@@ -1512,7 +1512,7 @@ int Fl_Wayland_Window_Driver::set_cursor_4args(const Fl_RGB_Image *rgb, int hotx
   // delete the previous custom cursor, if there was one, and keep its Fl_RGB_Image if appropriate
   delete_cursor_(xid, keep_copy);
   //have this new cursor used
-  xid->custom_cursor = new struct wld_window::custom_cursor;
+  xid->custom_cursor = new wld_window::custom_cursor_;
   xid->custom_cursor->wl_cursor = new_cursor;
   xid->custom_cursor->rgb = rgb;
   xid->custom_cursor->hotx = hotx;
```
After applying the patch, the test/cursor demo seems to work well.

**Comment by ManoloFLTK** [comment]
Thanks for the fix which is now committed.

**Comment by Albrecht-S** [comment]
Fixed the fix in commit 16338ff4e4e3a255a954d6efd61464d82196092a.
